vars/runATH.groovy

```diff
@@ -0,0 +1,206 @@
+            def rerunCount = metadata.rerunFailingTestsCount ?: 0
+            def localSnapshots = metadata.useLocalSnapshots ?: true
+
+            if (testsToRun == null && categoriesToRun == null) {
+                categoriesToRun = defaultCategory
+            }
+
+            def testingbranches = ["failFast": failFast]
+            for (browser in browsers) {
+                if (supportedBrowsers.contains(browser)) {
+
+                    def currentBrowser = browser
+                    def containerArgs = "-v /var/run/docker.sock:/var/run/docker.sock -e LOCAL_SNAPSHOTS=${localSnapshots} -e SHARED_DOCKER_SERVICE=true -u ath-user"
+                    def commandBase = "./run.sh ${currentBrowser} ./jenkins.war -Dmaven.test.failure.ignore=true -DforkCount=1 -B -Dsurefire.rerunFailingTestsCount=${rerunCount}"
+                    if (infra.isRunningOnJenkinsInfra()) {
+                        def settingsXml = "${pwd tmp: true}/settings-azure.xml"
+                        writeFile file: settingsXml, text: libraryResource('settings-azure.xml')
+                        commandBase = commandBase + " -s ${settingsXml}"
+                    }
+
+                    if (testsToRun) {
+                        testingbranches["ATH individual tests-${currentBrowser}"] = {
+                            dir("test${currentBrowser}") {
+                                unstash name: "athSources"
+                                unstash name: "jenkinsWar"
+                                def command = commandBase + " -Dtest=${testsToRun}"
+
+                                athContainerImage.inside(containerArgs) {
+                                    realtimeJUnit(testResults: 'target/surefire-reports/TEST-*.xml', testDataPublishers: [[$class: 'AttachmentPublisher']]) {
+                                        sh 'eval "$(./vnc.sh)" && ' + command
+                                    }
+
+                                }
+                            }
+                        }
+                    }
+                    if (categoriesToRun) {
+                        testingbranches["ATH categories-${currentBrowser}"] = {
+                            dir("categories${currentBrowser}") {
+                                unstash name: "athSources"
+                                unstash name: "jenkinsWar"
+                                def command = commandBase + " -Dgroups=${categoriesToRun}"
+                                athContainerImage.inside(containerArgs) {
+                                    realtimeJUnit(testResults: 'target/surefire-reports/TEST-*.xml', testDataPublishers: [[$class: 'AttachmentPublisher']]) {
+                                        sh 'eval "$(./vnc.sh)" && ' + command
+                                    }
+
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    echo "${browser} is not yet supported"
```

Comment:
Haven't we already assessed this in the beginning and defaulted to firefox?

Reply:
Ah wait, only if empty in line 51. Then I wonder if we should do all the validations there (empty and not supported)

Reply:
I thought about that, but this way is trivial to add support to new browsers in the future, just add it to the `supportedBrowsers` array and unsupported ones are still ignored without further code changes

Reply:
+1 on this approach